fastlane_bot/events/async_utils.py

```diff
@@ -12,8 +12,6 @@ def get_contract_chunks(contracts: List[Dict[str, Any]]) -> List[List[Dict[str,
 
 def get_abis_and_exchanges(mgr: Any) -> Dict[str, Any]:
     abis = {}
-    exchanges = {}
     for exchange in mgr.exchanges:
-        exchanges[exchange] = exchange_factory.get_exchange(key=exchange, cfg=mgr.cfg, exchange_initialized=False)
-        abis[exchange] = exchanges[exchange].get_abi()
+        abis[exchange] = mgr.exchanges[exchange].get_abi()
```

Comment:
This code looks extremely suspicious:
```
for exchange in mgr.exchanges:
    abis[exchange] = mgr.exchanges[exchange].get_abi()
```
First thing that comes to mind is that you may as well do:
```
for exchange in mgr.exchanges:
    abis[exchange] = exchange.get_abi()
```
But then the question that arises is - what exactly is `exchange`?
Because on the one hand, it is used as a key to the `abis` dictionary.
And on the other hand, it is used as an object calling one of its class methods (`get_abi`).

Python allows using objects as keys as far as I remember, but is this indeed what we're doing?
Have you verified that it works as desired?

Reply:
Your comment that you might as well do the following:

```python
for exchange in mgr.exchanges:
    abis[exchange] = exchange.get_abi()
```

 is not correct. The `mgr.exchanges` object is structured as follows, where `ExchangeA, ExchangeB...` are instances of the `Exchange` class.

```python    
mgr.exchanges = {"exchange_a": ExchangeA, "exchange_b": ExchangeB}
```

Thus, your comment implies calling the `.get_abi()` method on the string type key in the dictionary, which would obviously not work.

Reply:
Oh, heck, you're right.
The loop iterates the dictionary keys, not values.
So perhaps a better way to phrase it (though only for readability) would be:
```
for exchange_name, exchange in mgr.exchanges.items():
    abis[exchange_name] = exchange.get_abi()
```

Reply:
I just implemented your suggestion.